packages/labs/analyzer/src/lib/model.ts

```diff
@@ -176,3 +192,66 @@ export const getLitModules = (analysis: Package) => {
   }
   return modules;
 };
+
+export interface ReferenceInit {
+  name: string;
+  package?: string;
+  module?: string;
+  isGlobal?: boolean;
+}
+
+export class Reference {
+  readonly name: string;
+  readonly package: string | undefined;
+  readonly module: string | undefined;
+  readonly isGlobal: boolean;
+  constructor(init: ReferenceInit) {
+    this.name = init.name;
+    this.package = init.package;
+    this.module = init.module;
+    this.isGlobal = init.isGlobal ?? false;
+  }
+
+  get moduleSpecifier() {
+    return this.isGlobal
+      ? undefined
+      : this.package! + (this.module ? '/' + this.module : '');
+  }
+}
+
+export class Type {
+  type: ts.Type;
+  text: string;
+  references: Reference[];
+
+  constructor(type: ts.Type, text: string, references: Reference[]) {
+    this.type = type;
+    this.text = text;
+    this.references = references;
+  }
+}
+
+/**
+ * Returns a deduped / coalesced string of import statements required to load
+ * the given references.
+ */
+export const getImportsStringForReferences = (references: Reference[]) => {
```

Comment:
I wonder if you want to return an array of some Import object for further de-duping by the consumer

Reply:
Added a TODO. Let's cover that when we integrate it into the generators.